static/js/upload_widget.js

```diff
@@ -31,6 +34,17 @@ exports.build_widget = function (
         input_error.hide();
         clear_button.show();
         upload_button.hide();
+        const image_blob = URL.createObjectURL(file);
+        if (preview_link !== null) {
+            preview_link.show();
+            preview_link.on('click', function (e) {
+                // This click event must not get propagated to parent container otherwise the modal
+                // will not show up because of a call to `close_active_modal` in `settings.js`.
+                emoji_image.attr('src', image_blob);
+                emoji_preview.show();
+                e.preventDefault();
+            });
```

Comment:
Generally we do this sort of thing by creating a global click handler for CSS class used by the `preview_link`  object, rather than adding them individually on each link.

Reply:
Do we still need to move it to click handler? Given that there is no click for the preview.